Mathlib/RingTheory/Valuation/ValExtension.lean

```diff
@@ -0,0 +1,156 @@
+/-
+Copyright (c) 2024 Jiedong Jiang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiedong Jiang, Bichang Lei
+-/
+import Mathlib.RingTheory.Valuation.Integers
+import Mathlib.RingTheory.Ideal.LocalRing
+
+/-!
+# Extension of Valuation
+
+In this file, we define the typeclass for valuation extensions and prove basic facts about
+extension of valuations. Let `A` be an `R` algebra, equipped with valuation `vA` and `vR`
+respectively. Here, extension of valuation means that the pullback of valuation `vA` to `R` is
+equivalent to the valuation `vR` on `R`. We only require equivalence, not equality of valuations
+here.
+
+Note that we do not require the ring map from `R` to `A` to be injective. It holds automatically
+when `R` is a division ring and `A` is nontrivial.
+
+## Main Definition
+
+* `IsValExtension vR vA` : The valuation `vA` on `A` is an extension of the valuation `vR` on `R`.
+
+## References
+
+* [Bourbaki, Nicolas. *Commutative algebra*] Chapter VI §3, Valuations.
+* <https://en.wikipedia.org/wiki/Valuation_(algebra)#Extension_of_valuations>
+
+## Tags
+Valuation, Extension of Valuation
+
+-/
+open Valuation
+
+/--
+The class `IsValExtension R A` states that the valuation of `A` is an extension of the valuation
+on `R`. More precisely, the valuation on `R` is equivlent to the comap of the valuation on `A`.
+-/
+class IsValExtension {R A ΓR ΓA : Type*} [CommRing R] [Ring A]
+  [LinearOrderedCommMonoidWithZero ΓR] [LinearOrderedCommMonoidWithZero ΓA]
+  [Algebra R A] (vR : Valuation R ΓR) (vA : Valuation A ΓA) : Prop where
+  /-- The valuation on `R` is equivlent to the comap of the valuation on `A` -/
+  val_isEquiv_comap : vR.IsEquiv <| vA.comap (algebraMap R A)
+
+namespace IsValExtension
+
+section CoeLemma
+
+variable {R A ΓR ΓA : Type*} [CommRing R] [Ring A]
+    [LinearOrderedCommMonoidWithZero ΓR] [LinearOrderedCommMonoidWithZero ΓA]
+    [Algebra R A] (vR : Valuation R ΓR) (vA : Valuation A ΓA) [IsValExtension vR vA]
```

Comment:
Please consider moving this to L35. Then you don't need to repeat these assumptions on L40 and L100, etc.

Reply:
Thank you! Fixed. There is a tiny difference between `LinearOrderedCommMonoidWithZero` and `LinearOrderedCommGroupWithZero` which makes things not completely trivial.